maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java

```diff
@@ -261,8 +291,12 @@ private boolean isCompatible(PluginVersionRequest request, String version) {
 
         try {
             pluginManager.checkPrerequisites(pluginDescriptor);
-        } catch (Exception e) {
-            logger.warn("Ignoring incompatible plugin version {}", version, e);
+        } catch (PluginIncompatibleException e) {
+            if (logger.isDebugEnabled()) {
+                logger.warn("Ignoring incompatible plugin version {}: {}", version, e.getMessage(), e);
```

Comment:
This looks duplicate when you pass `e` already.

Reply:
Agreed, all I wanted to distinguish is "with stack trace" or "without stack trace".

Reply:
I know, but that would rather confuse when the message appears twice

Reply:
Well, mvn -X output is already quite confusing :smile: so unsure will _this_ confuse anyone... but agreed, will fix.

Reply:
done